underscore.js

```diff
@@ -1115,6 +1115,15 @@
     if (a === b) return a !== 0 || 1 / a === 1 / b;
     // A strict comparison is necessary because `null == undefined`.
     if (a == null || b == null) return a === b;
+    // `NaN`s are equivalent, but non-reflexive.
+    if (a !== a) return b !== b;
+    // Exhaust primitive checks
+    var type = typeof a;
+    if (type !== 'function' && type !== 'object' && typeof b !== 'object') return false;
```

Comment:
> I'm unsure if it's needed:

It just misses the early exit. Could be tackled and simplified with `_.isObject` use.